bam_mate.c

```diff
@@ -156,24 +156,28 @@ static bool plausibly_properly_paired(bam1_t* a, bam1_t* b)
         return false;
 }
 
-static void bam_format_cigar(const bam1_t* b, kstring_t* str)
+// Returns true on error, false on success
```

Comment:
I dislike the use of bool here, in particular true being error and false being success.

It looks like you're trying to return 0 (false) for success with non-zero being error in the traditional unix way, in which case it should be return 0 and not return false.  Using a bool just obfuscates things.

If you don't object I'll fix this and push back to the PR for you to comment on.